easse/samsa.py

```diff
@@ -261,21 +257,22 @@ def corpus_samsa(orig_sents: List[str], sys_sents: List[str], lowercase: bool =
     
     orig_sents = [utils_prep.normalize(sent, lowercase, tokenizer) for sent in orig_sents]
     orig_ucca_passages = ucca_parse_texts(orig_sents)
-    orig_synt_scenes = syntactic_parse_ucca_scenes(orig_ucca_passages, verbose=verbose)
+    orig_synt_scenes = syntactic_parse_ucca_scenes(orig_ucca_passages, tokenize=False, sentence_split=False,
+                                                   verbose=verbose)
 
     sys_sents = [utils_prep.normalize(output, lowercase, tokenizer) for output in sys_sents]
     sys_sents_synt = syntactic_parse_texts(sys_sents, tokenize=False, sentence_split=True, verbose=verbose)
 
-    samsa_score = 0.0
+    sentences_scores = []
     for orig_passage, orig_scenes, sys_synt in tqdm(zip(orig_ucca_passages, orig_synt_scenes, sys_sents_synt),
                                             disable=(not verbose)):
-        samsa_score += compute_samsa(orig_passage, orig_scenes, sys_synt)
+        sentences_scores.append(100. * compute_samsa(orig_passage, orig_scenes, sys_synt))
 
-    samsa_score /= len(orig_sents)
+    corpus_score = np.mean(sentences_scores)
 
-    return 100. * samsa_score
+    return sentences_scores, corpus_score
```

Comment:
I would prefer if we don't change the signature of this method for more uniformity with other methods named similarly.

Is it possible to split in two methods?
```
def get_samsa_sentence_scores(...):
    ...

def corpus_samsa(...):
    return np.mean(get_samsa_sentence_scores(...))
```

Reply:
what about adding a bool parameter `aggregate`: if True (default) will return the mean, if False will return the individual sentence scores?

Reply:
I would really prefer two methods, I think it's more readable.
I try to avoid using too many flags and `if`s in my functions in general, that's what make it harder to read and maintain. Ideally, to my mind, each function should represent one and one "logic" only and not be like a swiss army knife ^^

Reply:
ok, I made the change.